app/easyaccess.py

```diff
@@ -49,98 +62,173 @@ def new_file_slack_message(message, url, metadata):
                 {
                     "type": "button",
                     "text": "Open folder :open_file_folder:",
-                    "url": url_path,
+                    "url": dirname,
                 },
             ]
         }
     ]
-    duration = seconds_to_hms(metadata['duration_secs'])
-    formatted_message = "%s: %s (Duration %s)" % (message, filename, duration)
+
+    formatted_message = "%s: %s (Duration %s)" % (message, os.path.basename(url), duration)
     post_slack_message(formatted_message, attachments=attachments)
 
 
-def main():
-    logging.info("Looking for video files to convert in %s. Press Ctrl+C (several times) to quit." % settings.WATCH_FOLDER)
+def post_slack_exception(message):
+    traceback.print_exc()
+    logging.warning(message)
+    post_slack_message(message)
 
-    files = find_video_files(settings.WATCH_FOLDER)
 
-    # The transcoder exits after the first transcode. The service continually restarts in Balena, so we need only
-    # process the first file. This improves the scope for parallelisation since many transcoders will each just pick
-    # up the first unlocked file rather than looping through all files in parallel and potentially coming into
-    # conflict.
+def convert_and_get_metadata(source_file_path, dest_file_path, ffmpeg_base_args):
+    if os.path.exists(dest_file_path):
+        message = "Cancelling video conversion: " + dest_file_path + " already exists."
+        logging.warning(message)
+        post_slack_message(message)
+        return
 
+    with tempfile.TemporaryDirectory() as tmp_folder:
+        tmp_path = os.path.join(tmp_folder, os.path.basename(dest_file_path))
+        ffmpeg_args = ["ffmpeg", '-i', source_file_path] + ffmpeg_base_args + [tmp_path]
+        cmd_str = " ".join(ffmpeg_args)
+        logging.info("Running " + cmd_str)
+        r = subprocess.run(ffmpeg_args, check=True)
+        fixity_move(tmp_path, dest_file_path, failsafe_folder=None)
+        logging.info("Conversion complete: " + dest_file_path)
+
+    metadata = get_video_metadata(dest_file_path)
+    with open(dest_file_path + ".json", 'w') as f:
+        json.dump(metadata, f, indent=2, default=str)
+    vernon_id, file_type, title = parse_collection_file_path(dest_file_path)
+    metadata.update({'vernon_id': vernon_id, 'filetype': file_type, 'title': title})
+    write_metadata_summary_entry(metadata)
+    new_file_slack_message("*New file* :hatching_chick:", dest_file_path, seconds_to_hms(metadata['duration_secs']))
+
+    return metadata
+
+
+# The transcoder exits after the first transcode. The service continually restarts in Balena, so we need only
+# process the first file. This improves the scope for parallelisation since many transcoders will each just pick
+# up the first unlocked file rather than looping through all files in parallel and potentially coming into
+# conflict.
+def main():
```

Comment:
We may need to split this up into smaller functions in order to be able to test them more easily. This does not need to be done right now, it will most likely be done in ACMILabs/xos#1339.